dhall-docs/src/Dhall/Docs.hs

```diff
@@ -106,20 +119,32 @@ getAllDhallFilesAndHeaders
     :: Path Abs Dir -- ^ Base directory to do the search
     -> IO [(Path Abs File, Header)]
 getAllDhallFilesAndHeaders baseDir = do
-    files <- snd <$> Path.IO.listDirRecur baseDir
+    files <- filter hasDhallExtension . snd <$> Path.IO.listDirRecur baseDir
     Data.Maybe.catMaybes <$> mapM readDhall files
   where
+    hasDhallExtension :: Path Abs File -> Bool
+    hasDhallExtension absFile = case Path.splitExtension absFile of
```

Comment:
I somewhat wonder whether there might be a faster alternative to `splitExtensions` here, but it probably doesn't matter, at least for now.

Reply:
I guess that `fileExtension` is shorter, but it calls `splitExtension` under the hood as well:

https://github.com/commercialhaskell/path/blob/a75ddfc86319cd2187eb8965bdb11b52690fa002/src/Path/Include.hs#L459-L460

Reply:
Alternatively we could possibly use [`isSuffixOf`](http://hackage.haskell.org/package/base-4.14.0.0/docs/Data-List.html#v:isSuffixOf), but, again, I don't know whether that is actually faster, or whether we really _need_ anything faster.

Reply:
Oh, another solution can be to use `turtle` module to filter using `find`, check this example https://hackage.haskell.org/package/turtle-1.5.19/docs/Turtle-Prelude.html

The only downside is that it returns `FilePath`, and I won't like to do a lot of path conversions, that may uglify the code.

Also I don't know if that is actually faster. `splitExtension` implementation looks roughly O(n), and we execute it on each file. `find` executes the specified pattern on each file, but I'm not sure how much is the complexity difference. Maybe @Gabriel439 can illustrate us a little bit?

Another thing is that I tested with the biggest dhall package I found: dhall-kubernetes. And it isn't really that slow, so we can stick with `splitExtensions` with no problem

Reply:
I personally would stick with `splitExtension`.  It's highly unlikely that it would be the bottleneck

Reply:
@Gabriel439 Cool, then I'll mark this discussion as solved